client/python/gradio_client/client.py

```diff
@@ -184,13 +185,43 @@ def __init__(
         self.executor = concurrent.futures.ThreadPoolExecutor(max_workers=max_workers)
 
         # Disable telemetry by setting the env variable HF_HUB_DISABLE_TELEMETRY=1
-        threading.Thread(target=self._telemetry_thread).start()
+        threading.Thread(target=self._telemetry_thread, daemon=True).start()
+        self._refresh_heartbeat = threading.Event()
+        self._kill_heartbeat = threading.Event()
+
+        self.heartbeat = threading.Thread(target=self._stream_heartbeat, daemon=True)
+        self.heartbeat.start()
 
         self.stream_open = False
         self.streaming_future: Future | None = None
         self.pending_messages_per_event: dict[str, list[Message | None]] = {}
         self.pending_event_ids: set[str] = set()
 
+    def close(self):
+        self._kill_heartbeat.set()
+        self.heartbeat.join(timeout=1)
+
+    def _stream_heartbeat(self):
+        while True:
+            url = self.heartbeat_url.format(session_hash=self.session_hash)
```

Comment:
Will test this but this might lead to lots of cpu consumption. We may need to add a sleep

Reply:
Just tested and it will have a minimal impact on CPU. As long as the connection is open, the loop will be held up in `response.iter_lines()` which doesn't take up CPU as messages come every 15 seconds. We return to the top of the loop only when a new session is requested (`.reset_session`)